pubsub/subscriptions/detach_subscription.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_detach_subscription]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func detachSubscription(w io.Writer, projectID, subName string) error {
+	// projectID := "my-project-id"
+	// subName := "projects/my-project/subscriptions/my-sub"
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+
+	// Call DetachSubscription, which detaches a subscription from
+	// a topic you own. This can only be done if you have the
+	// pubsub.topics.detachSubscription role on the topic.
+	_, err = client.DetachSubscription(ctx, subName)
+	if err != nil {
+		return fmt.Errorf("detach subscription failed: %v", err)
+	}
+	fmt.Fprintf(w, "Detached subscription %s", subName)
```

Comment:
It will be good to show detached=true in the subscription after detachment.

Reply:
I would like to do this, but it causes the sample to look strange. `subName` is the fully qualified, globally unique name (required to call `DetachSubscription`). In the Go library, we expose two ways of grabbing a subscription, `client.Subscription` which uses the same `projectID` that was passed into the `NewClient` object. This isn't ideal because the subscription exist in another project. The alternative method is `client.SubscriptionInTopic`, which takes the project and subscription name separately (but requires splitting the subName before calling that method).

I check for `detached = true` in the tests, so I think that would suffice.